File: C++/calculadoraConFunciones.cpp
/*Calculadora hecho con Funciones*/
#include <iostream>
using namespace std;

int calculadora(int opcion, float num1, float num2, float solucion)
{

    cout << "1. Suma." << endl;
    cout << "2. Resta." << endl;
    cout << "3. Multiplicación." << endl;
    cout << "4. División." << endl;
    cout << "Elige entre el 1 - 4: ";
    cin >> opcion;

    cout << "Ingresa el primer número: ";
    cin >> num1;

    cout << "Ingresa el segundo número: ";
    cin >> num2;

    switch (opcion)
    {
    case 1:
        return num1 + num2;
        break;

    case 2:
        return num1 - num2;
        break;

    case 3:
        return num1 * num2;
        break;

    case 4:
        if (num2 == 0)
        {
            cout << "NO SE PUEDE DIVIDIR ENTRE CERO." << endl;
            break;
        }
        else
        {
            return num1 / num2;
        }

    default:
        cout << "Debes elegir entre el 1 - 4.";
        break;
    }
    return solucion;

    cout << ("Saliste de la calculadora.");
    return 0;
}

int main()

{
    int elegir;
    float ingresaPrimerNumero, ingresaSegundoNumero, desarrollo;
    cout << calculadora(elegir, ingresaPrimerNumero, ingresaSegundoNumero, desarrollo);
}

File: Python/calculadoraConFunciones.py
"""Calculadora hecho con Funciones"""

def calculadora():
    opcion = int(input("""1.Suma\n2.Resta.\n3. Multiplicación.\n4. División.\nElige entre 1 - 4: """))
    num1 = float(input("Ingresa el primer número: "))
    num2 = float(input("Ingresa el el segundo número: "))

    return elegir_case(opcion, num1, num2)

def elegir_case(opcion, num1, num2):
    if opcion == 1:
        resultado = num1 + num2
    elif opcion == 2:
        resultado = num1 - num2
    elif opcion == 3:
        resultado = num1 * num2
    elif opcion == 4:
        if num2 == 0:
            print("No se puede dividir entre CERO.")
        else:
            resultado = num1 / num2
    else:
        print("ERROR - Debes elegir entre 1 - 4")

    return resultado

solucion = calculadora()
if solucion:
    print(f"El resultado de la opreación: {solucion}")
else:
    print("No se pudo realizar la operacion")